sql/core/src/main/scala/org/apache/spark/status/api/v1/sql/SqlResource.scala

```diff
@@ -84,18 +119,40 @@ private[v1] class SqlResource extends BaseAppResource {
     }
 
     val duration = exec.completionTime.getOrElse(new Date()).getTime - exec.submissionTime
-    val planDetails = if (details) exec.physicalPlanDescription else ""
-    val metrics = if (details) printableMetrics(exec.metrics, exec.metricValues) else Seq.empty
+    val planDetails = if (details && planDescription) exec.physicalPlanDescription else ""
+    val metrics =
+      if (details) printableMetrics(exec.metrics, exec.metricValues, nodeIdAndWSCGIdMap)
+      else Seq.empty
     new ExecutionData(
       exec.executionId,
       status,
       exec.description,
       planDetails,
-      metrics,
       new Date(exec.submissionTime),
       duration,
       running,
       completed,
-      failed)
+      failed,
+      metrics)
   }
+
+  private def getNodeIdAndWSCGIdMap(graph: SparkPlanGraph): Map[Long, Option[Long]] = {
```

Comment:
how about just return 
```Map[Long, Long]```
to make it simpler?

Reply:
`WholeStageCodegenId` is being parsed through WSCG `nodeName`. This adds the protection at parsing level if it fails due to invalid WSCG `nodeName`. I think this `optional` protection can still be useful even if invalid `wholeStageCodegenId` points less cases.